### DISCUSSION TITLE
parse without a file

### DISCUSSION THREAD

**Discussion by haasdo95** [foonathan/lexy#41] [Q&A] [answered]
It is my understanding that currently lexy::parse only accepts a file returned by lexy::read_file. Is it possible to let lexy::parse accept a std::string?
I know this is not strictly necessary: this is a parser after all. But this feature will prove useful for unit testing. 
Thanks!

**Comment by foonathan** [comment, marked as answer]
The `lexy::parse` function accepts an arbitrary `Input`. The buffer returned by `lexy::read_file` is one, but there are multiple others: https://lexy.foonathan.net/reference/#input

You want the `lexy::string_input`: https://lexy.foonathan.net/reference/input/string_input/

```cpp
std::string str = ...;
auto result = lexy::parse<production>(lexy::string_input(str), …);
```
